externals/kyuubi-jdbc-engine/src/test/scala/org/apache/kyuubi/engine/jdbc/oracle/WithOracleContainer.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine.jdbc.oracle
+
+import com.dimafeng.testcontainers.OracleContainer
+import org.testcontainers.utility.DockerImageName
+
+import org.apache.kyuubi.engine.jdbc.WithJdbcServerContainer
+
+trait WithOracleContainer extends WithJdbcServerContainer {
+
+  private val oracleDockerImage = "gvenzl/oracle-xe:18.4.0-slim"
```

Comment:
what are the differences between oracle-free and oracle-xe? I see that [testcontainers providers both versions](https://testcontainers.com/modules/?s=Oracle), and Spark uses [gvenzl/oracle-free:23.5-slim](https://github.com/apache/spark/blob/master/connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleDatabaseOnDocker.scala)

Reply:
Thank you for your thorough reviews. Maybe there's little difference, I chose "gvenzl/oracle-xe:18.4.0-slim" as image because it's refered at [OracleSpec in testcontainer project](https://github.com/testcontainers/testcontainers-scala/blob/master/modules/oracle/src/test/scala/com/dimafeng/testcontainers/OracleSpec.scala). Both oracle-xe and oracle-free are free versions of Oracle Database, oracle-free is probably based on a newer version.

Reply:
the code was updated 3 yrs ago, I would like to followup the Spark behavior